tests/vmi_lifecycle_test.go

```diff
@@ -965,6 +967,29 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
 
 			})
 
+			It("the vmi with EVMCS HyperV feature should have correct hyperv and cpu features auto filled", func() {
+				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
+				vmi.Spec.Domain.Features = &v1.Features{
+					Hyperv: &v1.FeatureHyperv{
+						EVMCS: &v1.FeatureState{},
+					},
+				}
+
+				_, err := virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
+				Expect(err).ToNot(HaveOccurred(), "Should create VMI")
+
+				time.Sleep(time.Second * 5)
+				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
```

Comment:
@ksimon1 why do you need to wait 5 secs and also run HTTP GET for the VMI resource. I think you can read the VMI resource directly from the return value of the `Create`. The mutations occur on a webhook level, I don't think it's deferred, it should be "atomic" by means that the Create should return with the mutated VMI resource.

Reply:
I just test it and return value of the Create doesn't contain mutated vmi. I have to call another `Get` which returns mutated vmi.